From 722f4f74f7fa772c789d7693bd444d46ff782a2a Mon Sep 17 00:00:00 2001
From: Chen Jiahao <chenjiahao16@huawei.com>
Date: Thu, 23 Jun 2022 09:49:15 +0800
Subject: [PATCH 2679/4585] [Huawei] arm64: add dump instr before BUG in kernel

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187058

--------------------------------

In case of debugging need when undefined instructin exception
occurs, we should dump regs and the undefined instruction
before BUG in kernel mode. Which is usually used to compare
with objdump instructions.

Signed-off-by: Chen Jiahao <chenjiahao16@huawei.com>
---
 arch/arm64/kernel/traps.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/kernel/traps.c b/arch/arm64/kernel/traps.c
index 7c05bcf29013..f8ec533eb590 100644
--- a/arch/arm64/kernel/traps.c
+++ b/arch/arm64/kernel/traps.c
@@ -405,7 +405,9 @@ void do_undefinstr(struct pt_regs *regs)
 	if (call_undef_hook(regs) == 0)
 		return;
 
-	BUG_ON(!user_mode(regs));
+	if (!user_mode(regs))
+		die("Oops - undefined instruction", regs, 0);
+
 	force_signal_inject(SIGILL, ILL_ILLOPC, regs->pc, 0);
 }
 NOKPROBE_SYMBOL(do_undefinstr);
-- 
2.17.1

